src/entities/user.ts

```diff
@@ -1,50 +1,38 @@
-import { Entity, PrimaryGeneratedColumn, Column, BaseEntity } from 'typeorm';
+import { BaseEntity, Column, CreateDateColumn, Entity, Index, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';
 
 @Entity({ name: 'users' })
+@Index('UX_user_provider_provider_user_id', ['provider', 'providerUserId'], { unique: true })
 export class User extends BaseEntity {
-    @PrimaryGeneratedColumn('uuid')
+    @PrimaryGeneratedColumn('uuid', { primaryKeyConstraintName: 'PK_user_id' })
     id: string;
 
-    @Column({ unique: true })
+    @Index('IX_user_provider')
+    @Column({ name: 'provider' })
+    provider: string;
+
+    @Column({ name: 'provider_user_id' })
+    providerUserId: string;
+
+    @Index('UX_user_email', { unique: true })
+    @Column({ name: 'email' })
     email: string;
 
-    @Column({ nullable: true })
-    name: string;
+    @Column({ name: 'email_verified', default: false })
+    emailVerified: boolean;
 
-    @Column({ nullable: true })
-    givenName: string;
+    @Column({ name: 'given_name', nullable: true })
+    givenName?: string;
 
-    @Column({ nullable: true })
-    lastName: string;
+    @Column({ name: 'family_name', nullable: true })
+    familyName?: string;
```

Comment:
I think `given_name` and `family_name` are in the openid connect spec, so we might as well use those keys.

Reply:
E-mail, Name, Given Name and Family Name are all part of the OIDC profile spec and should be populated from the profile ideally.  Name isn't always a conjunction of Given Name and Family Name for exmaple

Reply:
Yeah this already happens for google, but we need higher perms for OneLogin to request  `https://vocab.account.gov.uk/v1/coreIdentityJWT` claim which includes names.

For now we've got enough to create an account.

No display name returned by OneLogin as far as I can see, so we might need to fudge one for now, hence the getter.